tests/build/build.go

```diff
@@ -79,14 +80,35 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build-ser
 			_, err = f.AsKubeAdmin.HasController.CreateApplication(applicationName, testNamespace)
 			Expect(err).NotTo(HaveOccurred())
 
-			componentName = fmt.Sprintf("%s-%s", "test-component-pac", util.GenerateRandomString(6))
-			pacBranchName = constants.PaCPullRequestBranchPrefix + componentName
+			customDefaultComponentName = fmt.Sprintf("%s-%s-%s", gitPrefix, "test-custom-default", util.GenerateRandomString(6))
+			customBranchComponentName = fmt.Sprintf("%s-%s-%s", gitPrefix, "test-custom-branch", util.GenerateRandomString(6))
+			pacBranchName = constants.PaCPullRequestBranchPrefix + customBranchComponentName
 			componentBaseBranchName = fmt.Sprintf("base-%s", util.GenerateRandomString(6))
 
 			err = f.AsKubeAdmin.CommonController.Github.CreateRef(helloWorldComponentGitSourceRepoName, helloWorldComponentDefaultBranch, helloWorldComponentRevision, componentBaseBranchName)
```

Comment:
Do we wrap this around with gitProvider switch here as well?

Reply:
Do we need to? It is ran multiple times for each provider as defined in [entries](https://github.com/konflux-ci/e2e-tests/blob/0a0050d4322521ab2eb4be2a2816666f0ab00b62/tests/build/build.go#L837). The `gitPrefix` is changed by Ginkgo based on which entry is the test running on

Reply:
when the gitPrefix is `gitlab`, wouldn't it create unnecessary branch again? I think we should call `f.AsKubeAdmin.CommonController.Github.CreateRef` or `f.AsKubeAdmin.CommonController.Gitlab.CreateBranch` based on value of `gitProvider` .